src/threadpool.c

```diff
@@ -261,6 +261,26 @@ static void init_once(void) {
   init_threads();
 }
 
+int uv_set_threadpool_size(unsigned int number) {
+  int r;
+  char num_str[12]; /* enough room for unsigned int and null terminator */
+
+  /* check if the threadpool is already initialized */
+  if (nthreads != 0)
+    return UV_EINVAL;
+
+  /* Set the new UV_THREADPOOL_SIZE */
+  sprintf(num_str, "%u", number);
+  r = uv_os_setenv("UV_THREADPOOL_SIZE", num_str);
+  if (r != 0)
+    return r;
```

Comment:
I'm not fully convinced about the "users are going to be checking the UV_THREADPOOL_SIZE env variable" argument. I'd rather add a function to get the number of threads as mentioned before. Also how is this going to work? I guess in order not to break anything `UV_THREADPOOL_SIZE` should take precedence over the API call? I don't know tbh.

Reply:
The nthreads variable will hold the value of the current number of threads. That is what I saw during development of this.

This is not breaking anything -as it is rn-, I considered mutating the env as a side effect as the easiest way to solve this and have predictable output. I’m not sure about having a different value for that env var and having a different number of available threads in the thread pool

Reply:
But that is always the case if the user sets the value in the env variable too late...

IMHO I wouldn't set the env variable. Just take the passed value and use it. The better @santigimeno proposed serves as a way to read the actual value.

Note there is another thing, IIRC we cap the value we read from the env var, so there is already precedent on the env var and reality being out of sync. We should do the same here, apply a cap, and not set the env var.